### PR TITLE
feat: add keda integration

### LINKED ISSUE #1057: [Feature]: Add integration for Keda
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

_No response_

### Solution Description

Add an integration for Keda, as well as an analyzer for ScaledObject resources.

### Benefits

This feature will allow for a first implementation of a new integration (that can be worked upon in the future) and a new analyzer for ScaledObjects to make use of this integration.

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by DragonAlex98**
Closes #1057 

## 📑 Description

This PR adds the integration for Keda and an analyzer for ScaledObject resources.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: pkg/integration/keda/keda.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		//TODO: All of this should be configurable
+		UpgradeCRDs:     true,
+		Wait:            false,
+		Timeout:         300,
+		CreateNamespace: true,
+	}
+
+	// Install a chart release.
+	// Note that helmclient.Options.Namespace should ideally match the namespace in chartSpec.Namespace.
+	if _, err := k.helm.InstallOrUpgradeChart(context.Background(), &chartSpec, nil); err != nil {
```
[the remark this answers is not in the document]
Comment (DragonAlex98) [reply pkg/integration/keda/keda.go:76]: I took inspiration for some of the code from the Trivy integration which, when activated, also installs an Helm chart.

File: pkg/analyzer/events.go
[the remark this answers is not in the document]
Comment (DragonAlex98) [reply pkg/analyzer/events.go:1]: If I was to use the FetchLatestEvent function from the analyzer package, I would get an "import cycle not allowed" error. I also thought 'util' would be a better fit for that function.

**Comment by JuHyung-Son** [comment]
checked it deployed nicely. lgtm.

**Review by JuHyung-Son — APPROVED** [review]
lgtm! greate contribute

**Pull request merged**